aoa-jupyter-lab

```diff
@@ -147,13 +144,35 @@ RUN echo "${NB_USER} ALL=(ALL) NOPASSWD: /usr/bin/java" >> /etc/sudoers
 #ENV REQUESTS_CA_BUNDLE=/opt/conda/ssl/cert.pem
 
 # Add AOA CLI config file
-RUN mkdir -p /home/jovyan/.aoa && \
-    echo 'aoa_credentials: YWRtaW46YWRtaW4=' >> /home/jovyan/.aoa/config.yaml && \
-    echo 'aoa_url: http://aoa-core:8080' >> /home/jovyan/.aoa/config.yaml   && \
-    echo 'auth_mode: basic' >> /home/jovyan/.aoa/config.yaml
+RUN mkdir -p /home/${NB_USER}/.aoa && \
+    echo 'aoa_credentials: YWRtaW46YWRtaW4=' >> /home/${NB_USER}/.aoa/config.yaml && \
+    echo 'aoa_url: http://aoa-core:8080' >> /home/${NB_USER}/.aoa/config.yaml   && \
+    echo 'auth_mode: basic' >> /home/${NB_USER}/.aoa/config.yaml
+
+# Fix permissions
+RUN fix-permissions "${CONDA_DIR}/ssl" && fix-permissions "/home/${NB_USER}/.aoa" && fix-permissions "/teradata" && chown -R ${NB_USER} "/home/${NB_USER}/work"
+
+# Remove default pyhon3 kernel
+RUN jupyter kernelspec uninstall -y python3
+
+# Create python 3.6 kernel
+RUN conda create -n py36 python=3.6 ipykernel
```

Comment:
we probably need to add some of the simple python libraries to each env? Not many, just like sklearn and pandas for example?

Reply:
sure, we can install those. Just have to double check as creating an env with conda, it already installs the same packages as those in the base env

Reply:
I don't think that it shares packages from the base env with other envs. If i create a new conda env and don't specify any packages, it should not have any outside of python AFAIK.. thats the point of it isn't it? to keep them separate?

 it would be better to install them as "conda run -n py37 pip install ..." anyway no? to avoid having some pip and some conda? we're just using conda for the env.

Also, what is the base conda env? is that used by jupyterlab as a kernel? If not, shouldn't we just wipe it for space? or how does all that work

Reply:
Base conda env is py3.9, is the python version that comes with JupyterLab v1.5 and it's where the lab executes. But we can pass the required packages as a txt list with conda create -n py36 python=3.6. The reason behind using conda is because in this case, it was the easiest way to create the envs with the python version. And conda is already installed in the jupyterlab.

Reply:
sorry, i think you miss-understood my comment. 100% it made sense to use conda like you have. 

Regarding the packages to install in each - feel free to do whatever way makes sense for you. I was just saying to use pip instead of conda within the conda env (so still using conda env but pip as the package manager) as the versions and even the package names are different between conda and pip sometimes

Reply:
Ok, cool. I'll play around a bit more, see which one is simplest and less problematic.